FluxNotes/flux
# [CCP] Integrate shim mapping functionality into flux notes

For [Jira 1814](https://jira.mitre.org/browse/ASCODCP-1814).

For @dehall's original smart on FHIR datasource, only the McOde family could load in the app because the Synthea FHIR for these patients had MCODE profiles. This PR allows for Synthea patients not in the McOde family (i.e.without the profiles) to load into the app by passing them through a mapper.

I wrote the [fhir-mapper library](https://github.com/standardhealth/fhir-mapper) that does the mapping of Synthea FHIR to MCODE v0.5. This library allows for other mappers to be written in the future, e.g. when we need to write a mapper for Epic@BWH FHIR to MCODE. This library is now part of Flux Notes.

Also created `GenericSmartOnFhirDstu2DataSource.jsx` that extends the original Smart on FHIR datasource and implements the mapping of the generic Synthea FHIR to MCODE.

To test:

1. `yarn install`
2. Open the [SMART App Launcher](http://moonshot-dev.mitre.org:4001/?auth_error=&fhir_version_1=r2&fhir_version_2=r2&iss=&launch_ehr=1&launch_url=http%3A%2F%2Flocalhost%3A3000%2Flaunchsynthea&patient=&prov_skip_auth=1&provider=&pt_skip_auth=1&public_key=&sb=&sde=&sim_ehr=0&token_lifetime=15&user_pt=) and click "Launch App"
3. Select a patient *not from the McOde family*. This patient should now load in the app similar to how paitients in the McOde family do on master since the mapping is done
4. Feel free to also review the code on [fhir-mapper](https://github.com/standardhealth/fhir-mapper) that does the actual mapping

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

**Running the Application**


**Task**


**Documentation**


**Code Quality**


**Tests**



## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code
